overlord/snapstate/policy.go

```diff
@@ -38,7 +38,12 @@ func inUseFor(deviceCtx DeviceContext) func(snap.Type) (boot.InUseFunc, error) {
 	}
 }
 
-func isCoreBootDevice(st *state.State) (bool, error) {
+type deviceTraits struct {
```

Comment:
this feels too close to DeviceContext? maybe we need a function that returns a fallback one instead? what am i missing?

Reply:
This is just a helper that extends the previous one a bit to extract an extra field from the device context. Code could be moved to the users - is you suggestion creating a new type that implements `DeviceContext` interface by returning default values?

Reply:
my worry is that we have already too many concepts in this space, see snap.Device and DeviceContext.  Can we do something else with DeviceCtx? should it return a fallback? does it need a mode/variant that can return a fallback?

Reply:
I've done 4fce2b4da51844744ba55b74e5e486873f6bbf14 but I am not fully convinced as the information it contains might not be always correct. Perhaps I should just remove the helper and do the checks in the user functions if that helps.

Reply:
I've reverted that change now and instead made sure that the system is seeded in snap deb -> snapd and ubuntu-core -> core transitions.

Reply:
can't the code then now use  DeviceCtx directly without needing this?

Reply:
Right, I've done this now